enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/lock/forseti/ForsetiClient.java

```diff
@@ -101,7 +101,7 @@
      */
     private final ExclusiveLock myExclusiveLock = new ExclusiveLock( this );
 
-    private boolean hasLocks;
+    private volatile boolean hasLocks;
```

Comment:
As far as I can see, this field is only ever access from one thread at a time, managed by our lock and transaction pools. So I don't see why it needs to be volatile.